src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.Arrays;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
```

Comment:
Consider whether `@BeforeAll` or `@BeforeEach` is more appropriate for initializing the `RegistrationService` instance. `@BeforeAll` runs once before all tests, while `@BeforeEach` runs before each test. If `RegistrationService` is stateless and thread-safe, `@BeforeAll` is suitable. If it's stateful or not thread-safe, use `@BeforeEach`.